presto-main/src/main/java/com/facebook/presto/operator/aggregation/BitwiseAndAggregation.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.NullableLongState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.StandardTypes;
+
+@AggregationFunction("bitwise_and_agg")
+public final class BitwiseAndAggregation
+{
+    private BitwiseAndAggregation() {}
+
+    @InputFunction
+    public static void bitAnd(NullableLongState state, @SqlType(StandardTypes.BIGINT) long value)
+    {
+        state.setNull(false);
+        state.setLong(state.getLong() & value);
```

Comment:
You might need to check if state is "null" before reading "state.getLong()". If anything, it will be 0, which will cause this to produce 0 for all values (you should figure out why tests don't catch this)